src/features/resources/resourcesSlice.ts

```diff
@@ -0,0 +1,611 @@
+          state.baseMoves = baseData.moves;
+          state.basePokemons = baseData.pokemons;
+          state.baseTypes = baseData.types;
+        }
+        state.status = Status.idle;
+      }),
+      builder.addCase(setAllDataAsync.rejected, (state) => {
+        state.status = Status.failed;
+      });
+    builder.addCase(sortingAsync.pending, (state) => {
+      state.status = Status.loading;
+    }),
+      builder.addCase(sortingAsync.fulfilled, (state, action) => {
+        const { payload } = action;
+        if (payload) {
+          const { data, resourceType } = payload;
+          const mapped = data?.mapped || [];
+          const base = data?.base || null;
+          if (isPokemon(resourceType)) {
+            state.currentPokemons = mapped as Array<PokemonDetails>;
+            state.pokemonSorting = PokemonSorting.none;
+            state.currentPokemonPage = 1;
+            state.basePokemons = base;
+          }
+          if (isMove(resourceType)) {
+            state.currentMoves = mapped as Array<PokemonMoveDetails>;
+            state.moveSorting = MoveSorting.none;
+            state.currentMovePage = 1;
+            state.baseMoves = base;
+          }
+          if (isType(resourceType)) {
+            state.currentTypes = mapped as Array<PokemonTypeDetails>;
+            state.typeSorting = TypeSorting.none;
+            state.currentTypePage = 1;
+            state.baseTypes = base;
+          }
+        }
+        state.status = Status.idle;
+      }),
+      builder.addCase(sortingAsync.rejected, (state) => {
+        state.status = Status.failed;
+      }),
+      builder.addCase(resultsAmountAsync.pending, (state) => {
+        state.status = Status.loading;
+      }),
+      builder.addCase(resultsAmountAsync.fulfilled, (state, action) => {
+        const { payload } = action;
+        if (payload) {
+          const { data, resourceType, resultsAmount } = payload;
+          const mapped = data?.mapped || [];
+          const base = data?.base || null;
+          if (isPokemon(resourceType)) {
+            state.currentPokemons = mapped as Array<PokemonDetails>;
+            state.pokemonResultsAmount = resultsAmount;
+            state.currentPokemonPage = 1;
+            state.basePokemons = base;
+          }
+          if (isMove(resourceType)) {
+            state.currentMoves = mapped as Array<PokemonMoveDetails>;
+            state.moveResultsAmount = resultsAmount;
+            state.currentMovePage = 1;
+            state.baseMoves = base;
+          }
+          if (isType(resourceType)) {
+            state.currentTypes = mapped as Array<PokemonTypeDetails>;
+            state.typeResultsAmount = resultsAmount;
+            state.currentTypePage = 1;
+            state.baseTypes = base;
+          }
+        }
+        state.status = Status.idle;
+      }),
+      builder.addCase(resultsAmountAsync.rejected, (state) => {
+        state.status = Status.failed;
```

Comment:
How you are planning to customize your request status notifications depending on operation if you use the same status field? Maybe it is worth to declare separate status fields for separate async operations?

Reply:
Yeah, I think it's nice to handle such situations. But in this app it's not necessary, I quess.